### ISSUE TITLE
[packagers] Codesign broken with homebrew formula on intel macos

### ISSUE THREAD

**Issue by jvalkeal** [jreleaser/jreleaser#1522]
Now, I'm on linux so talk to me like I'm 4 year old child what comes for macos.

It looks like issue is in these lines https://github.com/jreleaser/jreleaser/blob/ce9f4c587de3354cf304d67bf859374464ee87fe/core/jreleaser-templates/src/main/resources/META-INF/jreleaser/templates/jlink/brew/formula-multi.rb.tpl#L30-L31

I believe those were copied from rust formula https://github.com/Homebrew/homebrew-core/blob/d74c0bf3e0a30dd09cef8aa3b8acb5b3f3d0d887/Formula/r/rust.rb#L152-L159 There's also comment on https://github.com/Homebrew/homebrew-core/issues/106008#issuecomment-1186546624 that you can't just change dylib id without re-signing.

More info on `spring-cli` issue https://github.com/spring-projects-experimental/spring-cli/issues/105. In `spring-cli` we essentially started to use custom template and removed arch check so that codesign is done on both intel/arm. Colleagues having intel/arm confirmed brew formula started to work just fine.

### Steps to Reproduce

Install jreleaser from jreleaser/tap on macos intel.

```
# /usr/local/Cellar/jreleaser/1.8.0/libexec/bin
$ codesign --verify --verbose java
java: invalid signature (code or signature have been modified)
In architecture: x86_64
```

There's no issues in jreleaser installed from brew core.

### Expected Behaviour

Should run without errors

### Actual Behaviour

macos fails to run as codesign is invalid

### Environment Information

JReleaser 1.8.0 though we've used 1.9.0 snapshots as well.

```
HOMEBREW_VERSION: 4.1.18-29-g2b747f4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 2b747f45844bbd88c14c2c10812fa00e01261baf
Last commit: 3 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: ed85c609ca914b788fe07e68ff190287956477e1
Core tap last commit: 2 hours ago
Core tap branch: master
Core tap JSON: 31 Oct 15:02 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.6-x86_64
CLT: 15.0.0.0.1.1694021235
Xcode: N/A
```

**Comment by aalmiray** [comment]
In the meantime as a workaround you may generate a template and put the fix in place. Use the `template generate` command to get started.

**Comment by aalmiray** [comment]
It looks like this applies to distributions of type `JLINK` as they bundle a Java runtime which is re-signed on ARM.
Binaries should be resigned on Intel as well.

**Comment by aalmiray** [comment]
FWIW consuming `jreleaser` from brew core does not trigger this issue as that formula install the universal binary that does not bundle a Java Runtime.

**Comment by jvalkeal** [comment]
Yeah, that's what I meant when I mentioned that jreleaser from brew core works fine. I don't fully understand the issue in terms of if this is just specific to jreleaser/jlink. I was wondering if that rust formula in brew core has a same issue or if it just doesn't cause issues(as looks like signing code might have been copied from there).

**Comment by aalmiray** [comment]
As I understand the problem are the binaries provided by the bundled Java runtime, it's not a problem specific to Rust.

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.10.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.10.0))
